extensions-core/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureStorageDruidModule.java

```diff
@@ -117,25 +119,30 @@ public void configure(Binder binder)
 
   @Provides
   @LazySingleton
-  public CloudBlobClient getCloudBlobClient(final AzureAccountConfig config)
-      throws URISyntaxException, InvalidKeyException
+  public Supplier<CloudBlobClient> getCloudBlobClient(final AzureAccountConfig config)
```

Comment:
like previously mentioned
nit: I think leaving a comment here would help people understand that this is lazily supplied to avoid needing to set unnecessary properties